doc/source/numpy_2_0_migration_guide.rst

```diff
@@ -420,12 +420,14 @@ The :ref:`copy keyword behavior changes <copy-keyword-changes-2.0>` in
    needed" in a way that's compatible with NumPy 1.x and 2.x, see
    `scipy#20172 <https://github.com/scipy/scipy/pull/20172>`__ for an example
    of how to do so.
-3. For any ``__array__`` method on a non-NumPy array-like object, a
-   ``copy=None`` keyword can be added to the signature - this will work with
-   older NumPy versions as well. If ``copy`` keyword is considered in
-   the ``__array__`` method implementation, then for ``copy=True`` always
-   return a new copy.
-
+3. For any ``__array__`` method on a non-NumPy array-like object,
+   ``dtype=None`` and ``copy=None`` keywords must be added to the signature -
+   this will work with older NumPy versions as well (although older numpy versions
+   will never pass in these keywords). If the keywords are actually used in the ``__array__``
+   method implementation, then for:
+   * ``copy=True`` and any ``dtype`` value always return a new copy,
+   * ``copy=None`` create a copy if required (for example by ``dtype``),
+   * ``copy=False`` a copy must never be made. If ``dtype`` needs a copy then raise an exception.
```

Comment:
```suggestion
   * ``copy=False`` a copy must never be made. If a copy is needed to return a numpy array or satisfy ``dtype``, then raise an exception (``ValueError``).
```

Regardless of `dtype` a copy might be needed. And we can be more specific this should be a ValueError?

Reply:
Right - that's more precise. Done!